cmd/preflight/cmd/check_container.go

```diff
@@ -339,3 +352,74 @@ func artifactsTar(ctx context.Context, src string, w io.Writer) error {
 
 	return nil
 }
+
+func platformsToBeProcessed(cmd *cobra.Command, cfg *runtime.Config) ([]string, error) {
+	ctx := cmd.Context()
+	logger := logr.FromContextOrDiscard(ctx)
+
+	// flag.Changed is not set if the env is all that is set.
+	_, platformEnvPresent := os.LookupEnv("PFLT_PLATFORM")
+	platformChanged := cmd.Flags().Lookup("platform").Changed || platformEnvPresent
+
+	containerImagePlatforms := []string{cfg.Platform}
+
+	ref, err := name.ParseReference(cfg.Image)
+	if err != nil {
+		return nil, fmt.Errorf("invalid image reference: %w", err)
+	}
+
+	desc, err := remote.Get(ref)
+	if err != nil {
+		return nil, fmt.Errorf("invalid manifest?: %w", err)
+	}
+
+	if !desc.MediaType.IsIndex() && platformChanged {
+		// This means two things:
+		// 1) The user provided a platform, either via the cli or the env
+		// 2) The passed image is just an image, and not a manifest list
+		// Therefore, the given platform cannot be validated.
+		// --platform should only be passed with a manifest list.
+		return nil, fmt.Errorf("platform should not be specified on a manifest that is not an index")
+	}
+
+	// If platform param is not changed, it means that a platform was not specified on the
+	// command line. Therefore, we should process all platforms in the manifest list.
+	// As long as what is poinged to is a manifest list. Otherwise, it will just be the
+	// currnt runtime platform.
+	if desc.MediaType.IsIndex() {
+		logger.V(log.DBG).Info("manifest list detected, checking all platforms in manifest")
+
+		idx, err := desc.ImageIndex()
+		if err != nil {
+			return nil, fmt.Errorf("could not convert descriptor to index: %w", err)
+		}
+		manifestListDigest, err := idx.Digest()
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve index digest: %w", err)
+		}
+		cfg.ManifestListDigest = manifestListDigest.String()
+
+		manifest, err := idx.IndexManifest()
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve index manifest: %w", nil)
```

Comment:
This should be `err` not `nil`, otherwise `%w` won't work here.

Reply:
Oversight. Thanks.